torchvision/prototype/transforms/_geometry.py

```diff
@@ -0,0 +1,130 @@
+
+
+class Resize(Transform):
+    def __init__(
+        self,
+        size: Union[int, Tuple[int, int]],
+        *,
+        interpolation_mode: str = "nearest",
+    ) -> None:
+        super().__init__()
+        self.size = (size, size) if isinstance(size, int) else size
+        self.interpolation_mode = interpolation_mode
+
+    def get_params(self, sample: Any) -> Dict[str, Any]:
+        return dict(size=self.size, interpolation_mode=self.interpolation_mode)
+
+    @staticmethod
+    def image(input: Image, *, size: Tuple[int, int], interpolation_mode: str = "nearest") -> Image:
+        return Image(interpolate(input.unsqueeze(0), size, mode=interpolation_mode).squeeze(0), like=input)
+
+    @staticmethod
+    def bounding_box(input: BoundingBox, *, size: Tuple[int, int], **_: Any) -> BoundingBox:
+        old_height, old_width = input.image_size
+        new_height, new_width = size
+
+        height_scale = new_height / old_height
+        width_scale = new_width / old_width
+
+        old_x1, old_y1, old_x2, old_y2 = input.convert("xyxy").to_parts()
+
+        new_x1 = old_x1 * width_scale
+        new_y1 = old_y1 * height_scale
+
+        new_x2 = old_x2 * width_scale
+        new_y2 = old_y2 * height_scale
+
+        return BoundingBox.from_parts(new_x1, new_y1, new_x2, new_y2, like=input, format="xyxy", image_size=size)
+
+    def extra_repr(self) -> str:
+        extra_repr = f"size={self.size}"
+        if self.interpolation_mode != "bilinear":
+            extra_repr += f", interpolation_mode={self.interpolation_mode}"
+        return extra_repr
+
+
+class RandomResize(Transform, wraps=Resize):
+    def __init__(self, min_size: Union[int, Tuple[int, int]], max_size: Union[int, Tuple[int, int]]) -> None:
+        super().__init__()
+        self.min_size = (min_size, min_size) if isinstance(min_size, int) else min_size
+        self.max_size = (max_size, max_size) if isinstance(max_size, int) else max_size
+
+    def get_params(self, sample: Any) -> Dict[str, Any]:
+        min_height, min_width = self.min_size
+        max_height, max_width = self.max_size
+        height = int(torch.randint(min_height, max_height + 1, size=()))
+        width = int(torch.randint(min_width, max_width + 1, size=()))
+        return dict(size=(height, width))
+
+    def extra_repr(self) -> str:
+        return f"min_size={self.min_size}, max_size={self.max_size}"
+
+
+class Crop(Transform):
+    def __init__(self, crop_box: BoundingBox) -> None:
+        super().__init__()
+        self.crop_box = crop_box.convert("xyxy")
+
+    def get_params(self, sample: Any) -> Dict[str, Any]:
+        return dict(crop_box=self.crop_box)
+
+    @staticmethod
+    def image(input: Image, *, crop_box: BoundingBox) -> Image:
+        # FIXME: pad input in case it is smaller than crop_box
+        x1, y1, x2, y2 = crop_box.convert("xyxy").to_parts()
+        return Image(input[..., y1 : y2 + 1, x1 : x2 + 1], like=input)  # type: ignore[misc]
```

Comment:
At first I thought we weren't implementing it for `BoundingBox`. Might be good to maybe add a comment or something to indicate that `bounding_box` is handled by the default implementation.

Or maybe even better and be explicit about it and just implement a `bounding_box` method that just returns the identity?

Reply:
> At first I thought we weren't implementing it for `BoundingBox`. 

Your first impression is right: currently `Crop` is a no-op for `BoundingBox`'es. This is of course not "right" and needs to be implemented, but I haven't done that yet. What behavior to we want for this? Return `None` in case the area that the bounding box would cover after the crop is less than a threshold?

>  Might be good to maybe add a comment or something to indicate that `bounding_box` is handled by the default implementation.

What "default implementation" do you mean here? The default behavior is to ignore unknown `Feature`'s.

> Or maybe even better and be explicit about it and just implement a bounding_box method that just returns the identity?

Not sure the verbosity is worth it. Note for consistency we would need to do that for all unsupported `Feature`'s such as `Label` and so on. That means if we add a new `Feature` we need to touch every single transform to add a no-op to it.

IMO clearly communicating that "unlisted" feature transforms will be no-ops is easy enough to understand and reduces the boilerplate by quite a bit.

Reply:
I think we should either return the right thing, or raise a `NotImplemented` error.

This is one of the downsides of having a no-op by default -- no error is raised, and user might think that things are just magically working, which isn't the case.

Reply:
As discussed offline, we will go with explicit no-ops to avoid subtle bugs.